cmd/kubeadm/app/phases/bootstraptoken/node/token.go

```diff
@@ -105,8 +114,11 @@ func encodeTokenSecretData(tokenID, tokenSecret string, duration time.Duration,
 		data[bootstrapapi.BootstrapTokenDescriptionKey] = []byte(description)
 	}
 	for _, usage := range usages {
-		// TODO: Validate the usage string here before
+		// Validate the usage
+		if usage != "signing" && usage != "authentication" {
+			return nil, fmt.Errorf("Invalid usages, valid options: [signing,authentication]")
```

Comment:
Same code in two places. maybe make it as a function ?

Reply:
@wackxu Please move this code to `pkg/bootstrap/api` so it can be reused as @kad said.